### ISSUE TITLE
fatal error allowed memory size of 214743648 bytes exhausted (tried to allocate 4471112 bytes) in /cygdrive/........./Temp/............/.................../Crawler/Utils.php on line 638

### ISSUE THREAD

**Issue by chiragj-coder** [janreges/siteone-crawler#36]
![error output crawler stopped](https://github.com/user-attachments/assets/f9178cd7-31e3-4566-ae5a-6106ac7e66c1)

Same error happened two times.

![Basic Settings](https://github.com/user-attachments/assets/b6e0ba52-e61f-4fed-8af2-f19abc8d2320)
![Full Settings](https://github.com/user-attachments/assets/950c73af-547d-4c10-8648-4fb494f0b9e8)

**Comment by janreges** [comment]
Hi @chiragj-coder,

you have hit the default memory limit, which is set to 2GB. I recommend raising the memory limit to 4 - 16GB, depending on how much memory you have on your PC.

For example, if you have a PC with 16GB of RAM, you can set the `Memory limit` (see [docs](https://crawler.siteone.io/configuration/command-line-options/#--memory-limit)) item in the `Full settings` tab to `10G`. 

![image](https://github.com/user-attachments/assets/165709c7-6afc-43eb-8f20-fd3436602a12)

**Comment by chiragj-coder** [comment]
**Thanks for the quick response!**  

Just a small request:  

> Could you also display the default value in the app?  

This would make things clearer. I initially assumed that a blank field might mean "infinite" or "as much as available," as I've seen similar behavior in other software. While I did check the documentation, I found it a bit confusing because it mentions "2G," but the app shows a blank field.  

Thanks again for your support!
